packages/client/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,142 @@
+};
+
+type CheckboxInputProps = {
+  value: string;
+  filters: ObjType;
+  filtersName: string;
+  parentCallback: { name: string | null; value: string | null }[];
+};
+
+type ObjType = {
+  [key: string]: string;
+};
+
+type SelectProps = {
+  label: string;
+  selectObj: ObjType;
+};
+
+const BasicSelect = ({ label, selectObj }: SelectProps) => {
+  const [sort, setSort] = useState('');
+
+  const handleChange = (event: SelectChangeEvent<string>) => {
+    setSort(event.target?.value);
+  };
+
+  return (
+    <FormControl variant='standard' sx={{ minWidth: 120, mb: 2 }}>
+      <InputLabel id='simple-select-label'>{label}</InputLabel>
+      <Select labelId='simple-select-label' id='simple-select' value={sort} label={label} onChange={handleChange}>
+        {Object.keys(selectObj).map((key) => {
+          return (
+            <MenuItem key={key} value={selectObj[key]}>
+              {selectObj[key]}
+            </MenuItem>
+          );
+        })}
+      </Select>
+    </FormControl>
+  );
+};
+
+const CheckboxInput = ({ value, filters, filtersName, parentCallback }: CheckboxInputProps) => {
+  const [checked, setChecked] = useState(false);
+  function handleChecked() {
+    setChecked(!checked);
+  }
+
+  useEffect(() => {
+    if (parentCallback?.length === 0) {
+      setChecked(false);
+    }
+  }, [parentCallback]);
+
+  return (
+    <FormControlLabel
+      label={filters[value]}
+      control={<Checkbox onChange={() => handleChecked()} checked={checked} name={filtersName} value={value} />}
+    />
+  );
+};
+
+const CheckboxList = ({ listLabel, filters, filtersName, parentCallback }: CheckboxProps) => {
+  return (
+    <Box>
+      <FormControl margin='normal' component='fieldset' variant='standard'>
+        <FormLabel component='legend'>{listLabel}</FormLabel>
+        <FormGroup>
+          {Object.keys(filters).map((key: string) => {
+            return (
+              <CheckboxInput
+                parentCallback={parentCallback}
+                key={key}
+                value={key}
+                filters={filters}
+                filtersName={filtersName}
+              />
+            );
+          })}
+        </FormGroup>
+      </FormControl>
+    </Box>
+  );
+};
+
+const Filters = () => {
+  const [checked, setChecked] = useState<{ name: string | null; value: string | null }[]>([]);
+
+  const handleChecked = (e: ChangeEvent<HTMLInputElement>) => {
+    let updatedList = [...checked];
+
+    if (e.target?.getAttribute('type') === 'checkbox') {
+      if (e.target?.checked) {
+        updatedList = [...checked, { name: e.target.getAttribute('name'), value: e.target?.value }];
+      } else {
+        updatedList = updatedList.filter(
+          (item) => !_.isEqual(item, { name: e.target?.getAttribute('name'), value: e.target?.value }),
+        );
+      }
+    }
+    setChecked(updatedList);
+  };
+
+  const clearCheckboxes = () => {
+    setChecked([]);
+    return checked;
```

Comment:
Ta wartość zwracana jest do czegoś wykorzystywana? 🤔 jak nie to można usunąć

Reply:
miałam na początku inny zamysł i to pozostałość. Kasuję :)